### ISSUE TITLE
Chrome extension password clearly visible

### ISSUE THREAD

**Issue by poweredtreble** [NayamAmarshe/pashword#18]
Out of curiosity, why is the password clearly visible as soon as you click the Chrome Extension? Surely it should be hidden, at the very least, since the crux of the whole matter IS the master password, without any option to "show hidden password" etc.

You don't want a bad actor in your surroundings to easily gain access to the master password.

**Comment by NayamAmarshe** [comment]
Hi, I'll add an option to show/hide it :) Thanks for reporting!

**Comment by poweredtriple** [comment]
Hmm, I feel as if there shouldn't be an option to show it at all. it should be hidden by default, without any option of viewing it. A person should know their master password really well.

And perhaps password not saved by browser, but deleted securely on browser close? Or at least an option in the extension to have the master password deleted every time the browser is closed.

Sorry, I'm not a programmer, just thinking about it from a layman's understanding of security.

**Comment by cleverestx** [comment]
> Hmm, I feel as if there shouldn't be an option to show it at all. it should be hidden by default, without any option of viewing it. A person should know their master password really well.
> 
> And perhaps password not saved by browser, but deleted securely on browser close? Or at least an option in the extension to have the master password deleted every time the browser is closed.
> 
> Sorry, I'm not a programmer, just thinking about it from a layman's understanding of security.

I disagree with this. Please give us option to show it (not enabled by default though), as I have a crazy long password for my master password despite knowing it well; if I fumble type it I don't want to tart over...it helps if I can check if I typed it correctly. I am alone in my space where this is installed, and not having this feature would just hurt the experience for me.

**Comment by poweredtreble** [comment]
> I disagree with this. Please give us option to show it (not enabled by default though), as I have a crazy long password for my master password despite knowing it well; if I fumble type it I don't want to tart over...it helps if I can check if I typed it correctly. I am alone in my space where this is installed, and not having this feature would just hurt the experience for me.

Hmm I can see where you're coming from. As with many things related to security, you have to strike a balance between security and convenience.

Perhaps a good compromise could be an option to enable/disable this feature? So that users who are NOT alone in their space can disable the feature of viewing the master password. At the end of the day, from a security perspective, I believe the extension should be created with the assumption that there ARE bad actors in the users' surroundings.

(I think this is what you already suggested, I'm just making sure.)

**Comment by cleverestx** [comment]
Yeah, that's fine with me. Just give me the option in my own Docker installed local instance to allow that feature to exist...in the same way you disallow it another... gives more power to the users.

**Comment by NullPlane** [comment]
_Hardcodedly force-forbidding a feature out of privacy concerns and therefore reducing QOL is a security flaw itself_  
_as it is unintentionally subconsciously pushing people away from staying safe into acting more risk-taking._

Many people perform ten finger typing and even the pros are never 100% accurate. If you do not want to have a 
typo, people tend to do two things:
- type their MPW elsewhere and then copy-paste it which takes up even more time and gives  
  larger attack surface, 
- typing by Hunt-and-Peck, which significantly reduces their typing speed and therefore gives  
  any watcher greater chances to read the MPW off the input device

Giving people the possibility to quickly view their integrity of the master PW is a crucial feature I personally think.
Just imagine being in a situation you need quick access on an alien system or someone other's where you are not
allowed to use password managers. 



The best way I think is to implement an unhide-button that does not allow for static cached unhiding but rather
only allows unhiding the master password as long as a click on it is held pressed. Therefore quickly clicking it to 
check a specific section of your MPW is possible and gives no time for any watcher to perceive (except your screen
is recorded with a high enough fps but in that case I'd rather not use the system at all) 

A partial unhiding technique would also fit but is rather complex and useless in the end. But still, yes, an option to 
disable this feature may also exist for people in custom environments or anyone else who wants to; it is a legitimate 
suggestion too.
